avocado/utils/vmimage.py

```diff
@@ -0,0 +1,251 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2017
+# Author: Amador Pahim <apahim@redhat.com>
+
+"""
+Provides VM images acquired from official repositories
+"""
+
+import os
+import tempfile
+import uuid
+
+from . import asset
+from . import path as utils_path
+from . import process
+
+try:
+    import lzma
+    LZMA_CAPABLE = True
+except ImportError:
+    LZMA_CAPABLE = False
+
+
+class VMImageError(Exception):
+    """
+    Generic error class for VM Image exceptions
+    """
+
+
+class Image(object):
+
+    def __init__(self, name, major, minor, build, arch, checksum, algorithm,
```

Comment:
This number of mandatory arguments, even for a base class, bothers me.  Maybe have a `ImageSpec` class instead?  Where all (but maybe name) are optional?

```python
class ImageSpec(object):
   """
   Specifies a selector for a given image type
   """
   def __init__(self, name, major=None, minor=None, ...)
   ...
```

That would prompt the current `Image` class to become more of a `ImageProvider`, that knows how to get the image for a given "name".